src/main/java/org/commonjava/util/partyline/impl/infinispan/model/InfinispanJFS.java

```diff
@@ -1,659 +1,132 @@
-                prevBlock = currBlock;
-                currBlock = block;
-
-                flush();
-
-                currBlock.writeToBuffer ( (byte) b );
-            }
-        }
-
-        public void write( ByteBuffer buf )
-                throws IOException
-        {
-            while ( buf.hasRemaining() )
-            {
-                byte b = buf.get();
-                write( (int) b);
-            }
-        }
-
-        /**
-         * Empty the current buffer into the {@link FileChannel} and reinitialize it for filling. Increment the flushed-byte count, which is used as the
-         * read limit for associated {@link JoinInputStream}s. Notify anyone listening that there is new content via {@link JoinableFile#notifyAll()}.
-         */
-        @Override
-        public void flush()
-                throws IOException
-        {
-            synchronized ( InfinispanJFS.this )
-            {
-                if ( closed )
-                {
-                    throw new IOException( "Cannot write to closed stream!" );
-                }
-            }
-
-            doFlush( false );
-        }
-
-        @Override
-        public void close()
-                throws IOException
-        {
-            Logger logger = LoggerFactory.getLogger( getClass() );
-            logger.trace( "OUT ({}):: close() called", originalThreadName );
-
-            if ( closed )
-            {
-                logger.trace( "OUT ({}):: already closed", originalThreadName );
-                return;
-            }
-            doFlush( true );
-            super.close();
-
-            InfinispanJFS.this.close();
-        }
-
-        public void doFlush( boolean eof )
-                throws IOException
-        {
-            if ( eof )
-            {
-                filesystem.pushNextBlock( prevBlock, currBlock, metadata );
-            }
-            else
-            {
-                filesystem.updateBlock( prevBlock );
-            }
-
-            synchronized ( InfinispanJFS.this )
-            {
-                InfinispanJFS.this.notifyAll();
-            }
-
-            if ( callbacks != null )
-            {
-                callbacks.flushed();
-            }
-        }
-
-        boolean isClosed() {
-            return closed;
+            throw new IOException( "Thread interrupted while closing file: " + metadata.getFilePath(), e );
         }
     }
 
-    /**
-     * {@link InputStream} associated with a particular {@link JoinableFile} instance. This stream reads content that the output stream has
-     * already flushed to disk, and waits for new content to become available (or for the output stream to close). This allows multiple readers
-     * when content is still being written to disk.
-     */
-    private final class JoinInputStream
-            extends InputStream
+    FileMeta getMetadata( final File target, final LocalLockOwner owner ) throws IOException
```

Comment:
We should change this to have a boolean parameter "create". We should only create entries if this parameter is true. We should be able to handle that in the computeIfAbsent call, I think.

Eventually we will need to implement a call like File.exists() in partyline, on the cluster. Avoiding the creation of a new FileMeta on an existence check will be important to this, I think.